velox/type/StringView.h

```diff
@@ -101,7 +113,8 @@ struct StringView {
     return size <= kInlineSize;
   }
 
-  const char* data() const {
+  const char* data() && = delete;
```

Comment:
is this to give a compilation error if clients try to move it out?

Reply:
This accessor is not safe for inline string views. For example, `auto ptr = StringView("hellO").data()` produces a dangling pointer into the temporary. The same is true for other accessors like `begin()`.